lib/srv/authhandlers.go

```diff
@@ -357,9 +379,45 @@ func (h *AuthHandlers) UserKeyAuth(conn ssh.ConnMetadata, key ssh.PublicKey) (*s
 		return nil, trace.Wrap(err)
 	}
 
+	if err := maybeAppendDiagnosticTrace(ctx, connectionDiagnosticID, h.c.AccessPoint,
+		types.ConnectionDiagnosticTrace_CONNECTIVITY,
+		"Node is alive and reachable.",
+		nil,
+	); err != nil {
+		fmt.Println(err)
+		return nil, trace.Wrap(err)
+	}
+
+	if err := maybeAppendDiagnosticTrace(ctx, connectionDiagnosticID, h.c.AccessPoint,
+		types.ConnectionDiagnosticTrace_RBAC_PRINCIPAL,
+		"Successfully authenticated.",
+		nil,
+	); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
 	return permissions, nil
 }
 
+func maybeAppendDiagnosticTrace(ctx context.Context, connectionDiagnosticID string, ap AccessPoint, traceType types.ConnectionDiagnosticTrace_TraceType, message string, traceError error) error {
+	if connectionDiagnosticID == "" {
+		return nil
+	}
+
+	connectionTrace := types.NewSuccessTraceConnectionDiagnostic(traceType, message)
+
+	if traceError != nil {
+		connectionTrace = types.NewFailedTraceConnectionDiagnostic(
+			traceType,
+			message,
+			traceError,
+		)
+	}
```

Comment:
Nit: I'd make `types.NewTraceConnectionDiagnostic` method that accepts type, message and error and returns either success or failure trace depending on whether error is nil.